crates/noirc_evaluator/src/ssa/optimizations.rs

```diff
@@ -98,6 +99,49 @@ fn evaluate_intrinsic(
             }
             unreachable!();
         }
+        builtin::Opcode::ToRadix(endian) => {
+            let mut element = args[0].to_le_bytes().to_vec();
+            if endian == Endian::Big {
+                element = args[0].to_be_bytes().to_vec();
+            }
+            let byte_count = args[2] as u32;
+            let diff = if byte_count > element.len() as u32 {
+                byte_count - element.len() as u32
+            } else {
+                0
+            };
+            if endian == Endian::Little {
+                element.extend(vec![0; diff as usize])
+            } else {
+                let mut new_element = vec![0; diff as usize];
+                new_element.extend(element);
+                element = new_element;
+            }
+            let mut result = Vec::new();
+
+            if let ObjectType::Pointer(a) = res_type {
+                for (i, item) in element.iter().enumerate() {
+                    let index = ctx.get_or_create_const(
+                        FieldElement::from(i as i128),
+                        ObjectType::NativeField,
+                    );
+                    let op = if *item == 0 {
```

Comment:
you do not need to distinguish the zero case, so you can simply keep the else case here. Note that using ctx.zero() is wrong as it is not a FieldElement.